marimo/_ast/app.py

```diff
@@ -143,18 +143,23 @@ def __init__(self, **kwargs: Any) -> None:
         # a TypeError.
         self._config = _AppConfig.from_untrusted_dict(kwargs)
 
-        self._cell_manager = CellManager()
+        try:
+            # nested applications get a unique cell prefix to disambiguate
+            # their graph from other graphs
+            get_context()
+            cell_prefix = str(uuid4())
+        except ContextNotInitializedError:
```

Comment:
i've seen `except ContextNotInitializedError:` in a lot of places. maybe we could just create a util `if has_notebook_context()`, so its a bit more human readable

Reply:
done